src/awkward/contents/unionarray.py

```diff
@@ -614,18 +615,15 @@ def _nested_tags_index(self, offsets: Index, counts: Sequence[Index]):
         )
 
     def _getitem_next_jagged_generic(self, slicestarts, slicestops, slicecontent, tail):
-        simplified = self.simplify_uniontype()
-        if isinstance(simplified, ak.contents.UnionArray):
+        if isinstance(self, ak.contents.UnionArray):
             raise ak._errors.index_error(
                 self,
                 ak.contents.ListArray(
                     slicestarts, slicestops, slicecontent, parameters=None
                 ),
                 "cannot apply jagged slices to irreducible union arrays",
             )
-        return simplified._getitem_next_jagged(
-            slicestarts, slicestops, slicecontent, tail
-        )
+        return self._getitem_next_jagged(slicestarts, slicestops, slicecontent, tail)
```

Comment:
This one is not a `simplify_uniontype` → `simplified` pair because it's sanitizing input. There was a test providing insane inputs, so the `simplified` part of the pair went into the testing code.